sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -76,164 +89,303 @@ public FormRecognizerServiceVersion getServiceVersion() {
+     * @param formContentType Supported Media types including .pdf, .jpg, .png or .tiff type file stream.
      *
      * @return A {@link PollerFlux} that polls the extract receipt operation until it has completed, has failed, or has
      * been cancelled.
      */
-    public PollerFlux<OperationResult, IterableStream<ExtractedReceipt>> beginExtractReceiptsFromUrl(String sourceUrl) {
-        return beginExtractReceiptsFromUrl(sourceUrl, false, null);
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<OperationResult, IterableStream<RecognizedForm>>
+        beginRecognizeCustomForms(Flux<ByteBuffer> data, String modelId, long length, FormContentType formContentType) {
+        return beginRecognizeCustomForms(data, modelId, length, formContentType, false, null);
     }
 
     /**
-     * Detects and extracts data from receipts using optical character recognition (OCR) and a prebuilt receipt trained
+     * Recognizes and extracts form data from documents using optical character recognition (OCR) and a custom trained
      * model.
      * <p>The service does not support cancellation of the long running operation and returns with an
      * error message indicating absence of cancellation support.</p>
      *
-     * @param sourceUrl The source URL to the input document. Size of the file must be less than 20 MB.
+     * @param data The data of the document to be extract receipt information from.
+     * @param modelId The UUID string format custom trained model Id to be used.
+     * @param length The exact length of the data. Size of the file must be less than 50 MB.
      * @param includeTextDetails Include text lines and element references in the result.
+     * @param formContentType Supported Media types including .pdf, .jpg, .png or .tiff type file stream.
      * @param pollInterval Duration between each poll for the operation status. If none is specified, a default of
      * 5 seconds is used.
      *
      * @return A {@link PollerFlux} that polls the extract receipt operation until it has completed, has failed, or has
      * been cancelled.
      */
-    public PollerFlux<OperationResult, IterableStream<ExtractedReceipt>>
-        beginExtractReceiptsFromUrl(String sourceUrl, boolean includeTextDetails,
-                                Duration pollInterval) {
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<OperationResult, IterableStream<RecognizedForm>>
+        beginRecognizeCustomForms(Flux<ByteBuffer> data, String modelId, long length, FormContentType formContentType,
+        boolean includeTextDetails, Duration pollInterval) {
+        Objects.requireNonNull(data, "'data' is required and cannot be null.");
+        Objects.requireNonNull(modelId, "'modelId' is required and cannot be null.");
+        Objects.requireNonNull(formContentType, "'formContentType' is required and cannot be null.");
```

Comment:
Throwing exception in a method returning a reactor publisher will cause issues downstream. Instead all these exceptions have to be wrapped and returned in error channel.